### DISCUSSION TITLE
custom opengraph element for each post

### DISCUSSION THREAD

**Discussion by maguzzi** [hexojs/hexo#5346] [Question / Answer] [answered]
Is it possible to have custom opengraph image, title (for social preview) for each posts?
I'm new to nodejs and hexo. 
I tried implementing a custom plugin but it's difficult to figure out if the code is actually executed or not
Thanks!

**Comment by yoshinorin** [comment]
@maguzzi 
You can use [open-graph helper](https://hexo.io/docs/helpers#open-graph) in your template engines.  (e.g. [hexo-theme-landsacepe](https://github.com/hexojs/hexo-theme-landscape/blob/116d6ab200dbe0215dd1698859ac4b29be25f0ee/layout/_partial/head.ejs#L23))

Many themes probably creates og for title, image and etc... by default, but if they don't outputs as you want, you may need to customize the theme's template engine.

**Comment by maguzzi** [comment, marked as answer]
Hi @yoshinorin
thanks a lot for your answer. The actual solution was way easier: it was enough to remove the fixed image and urls I've put into _config.yaml, and those are now resolved on the fly for each post. I've replied in order not to make confusion in further readers
Bye
